sdk/azidentity/managed_identity_credential.go

```diff
@@ -11,9 +11,25 @@ import (
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
 )
 
+// IDKind is used to specify the type of identifier that is passed in for a user-assigned managed identity.
+type IDKind int
```

Comment:
I think IDKind might be too generic sounding. Perhaps ManagedIdentityIDKind or maybe just ManagedIdentityKind. I'm trying to draw a tight relationship that this is used specifically with the ManagedIdentityCredential type.

Reply:
My vote is for `ManagedIdentityKind`

Reply:
I was actually leaning towards `ManagedIdentityIDKind` because `ManagedIdentityKind` implies that this is saying what type of managed identity they're choosing instead of the type of identifier. Although I think it is very long. I'm trying to think of other possibilities too...maybe something related to the fact that this is something used with user assigned identities.

Reply:
I think I'll take `ManagedIdentityIDKind`. I agree that it's better to leave it more general than my idea to specify something related to user assigned identities. 👍

Reply:
Done.